e2e/helpers.ts

```diff
@@ -141,13 +141,51 @@ export const uploadImageToSection = async ({
   await uploadImage(page, buttonLocator)
 }
 
-export const expectTextWithChangeLink = async (
+export const expectAddress = async (
   locator: Locator,
-  texts: string[]
+  address: { [key: string]: any },
+  isDeletion?: boolean
 ) => {
+  const addressKeys = [
+    'country',
+    'province',
+    'district',
+    'village',
+    'town',
+    'residentialArea',
+    'street',
+    'number',
+    'addressLine1',
+    'addressLine2',
+    'addressLine3',
+    'postcodeOrZip',
+    'zipCode'
+  ]
+
+  const texts = addressKeys
+    .map((key) => address[key])
+    .filter((value) => Boolean(value))
+
+  if (isDeletion) {
+    for (let i = 0; i < texts.length; i++) {
+      await expect(
+        locator.getByRole('deletion').nth(i + (i < 3 ? 1 : 2))
```

Comment:
Since this involves numerical logic could we add a comment to clarify what is happening